javatests/arcs/core/host/ReflectiveParticleConstructionTest.kt

```diff
@@ -0,0 +1,73 @@
+package arcs.core.host
+
+import arcs.core.allocator.Allocator
+import arcs.core.data.FieldType
+import arcs.core.data.Plan
+import arcs.core.storage.driver.RamDisk
+import arcs.core.storage.driver.RamDiskDriverProvider
+import arcs.core.storage.keys.RamDiskStorageKey
+import arcs.core.storage.keys.VolatileStorageKey
+import arcs.jvm.host.ExplicitHostRegistry
+import arcs.jvm.host.JvmHost
+import arcs.jvm.host.JvmSchedulerProvider
+import arcs.jvm.util.testutil.FakeTime
+import com.google.common.truth.Truth.assertThat
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.runBlocking
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+import kotlin.coroutines.EmptyCoroutineContext
+
+
+@RunWith(JUnit4::class)
+@ExperimentalCoroutinesApi
+class ReflectiveParticleConstructionTest {
+
+    class JvmProdHost(
+        schedulerProvider: SchedulerProvider,
+        vararg particles: ParticleRegistration
+    ) : JvmHost(schedulerProvider, *particles), ProdHost
+
+    class AssertingPlannedParticle(spec: Plan.Particle?) : TestPlannedParticle(spec) {
+        override suspend fun onCreate() = runBlocking {
+            assertThat(schema.name?.name).isEqualTo("Thing")
```

Comment:
add companion object { var started = false }

and then assign started = true inside onCreate()